ignite/handlers/param_scheduler.py

```diff
@@ -1393,6 +1393,133 @@ def simulate_values(cls, num_events: int, schedulers: List[_LRScheduler], **kwar
+            metric monitored has stopped increasing. Default: 'min'.
+        factor: Factor by which the learning rate will be
+            reduced. new_lr = lr * factor. Default: 0.1.
+        patience: Number of epochs with no improvement after
+            which learning rate will be reduced. For example, if
+            `patience = 2`, then we will ignore the first 2 epochs
+            with no improvement, and will only decrease the LR after the
+            3rd epoch if the loss still hasn't improved then.
+            Default: 10.
+        threshold: Threshold for measuring the new optimum,
+            to only focus on significant changes. Default: 1e-4.
+        threshold_mode: One of `rel`, `abs`. In `rel` mode,
+            dynamic_threshold = best * ( 1 + threshold ) in 'max'
+            mode or best * ( 1 - threshold ) in `min` mode.
+            In `abs` mode, dynamic_threshold = best + threshold in
+            `max` mode or best - threshold in `min` mode. Default: 'rel'.
+        cooldown: Number of epochs to wait before resuming
+            normal operation after lr has been reduced. Default: 0.
+        min_lr: A scalar or a list of scalars. A
+            lower bound on the learning rate of all param groups
+            or each group respectively. Default: 0.
+        eps: Minimal decay applied to lr. If the difference
+            between new and old lr is smaller than eps, the update is
+            ignored. Default: 1e-8.
+        trainer: Trainer engine to log LR history in its
+            `state.output.param_history`. Is used if `save_history`
+            is true. Default: None.
+        save_history: Whether to save history or not. If true,
+            history will be logged in `trainer`'s `state.output.param_history`.
+            Default: False.
+        param_group_index: `optimizer`'s parameters group
+            to use.  Default: None. Use all `optimizer`'s paramater groups.
+
+    Examples:
+
+        .. code-block:: python
+
+            # Metric 'metric-name' should surpass its best value by
+            # more than 1 unit after at most 2 epochs, otherwise LR
+            # would get multiplied by 0.5 .
+
+            scheduler = ReduceLROnPlateauScheduler(
+                optimizer,
+                metric_name="metric-name", mode="max",
+                factor=0.5, patience=1, threshold_mode='abs',
+                threshold=1, trainer=trainer
+            )
+
+            evaluator.add_event_handler(Events.COMPLETED, scheduler)
+
+    """
+
+    def __init__(
+        self,
+        optimizer: Optimizer,
+        metric_name: str,
+        mode="min",
+        factor: float = 0.1,
+        patience: int = 10,
+        threshold: float = 1e-4,
+        threshold_mode: str = "rel",
+        cooldown: int = 0,
+        min_lr: Union[float, List[float]] = 0.0,
+        eps: float = 1e-8,
+        trainer: Engine = None,
+        save_history: bool = False,
+        param_group_index: int = None,
+    ):
```

Comment:
I think we can make constructor more simple. Keep only optimizer, trainer, save_history, param_group_index. Remove others and add **scheduler_kwargs which is passed to `ReduceLROnPlateau`. In docs we explain that **scheduler_kwargs are args/kwargs for `ReduceLROnPlateau`